src/tlo/events.py

```diff
@@ -63,8 +65,54 @@ def apply(self, target):
 
     def run(self):
         """Make the event happen."""
+        
+        print_chains = False
+        df_before = []
+        
+        if self.sim.generate_event_chains:
+            # Only print event if it belongs to modules of interest and if it is not in the list of events to ignore
+            if (self.module in self.sim.generate_event_chains_modules_of_interest) and all(sub not in str(self) for sub in self.sim.generate_event_chains_ignore_events):
+                print_chains = True
+                if self.target != self.sim.population:
+                    row = self.sim.population.props.iloc[[self.target]]
+                    row['person_ID'] = self.target
+                    row['event'] = self
+                    row['event_date'] = self.sim.date
+                    row['when'] = 'Before'
+                    self.sim.event_chains = pd.concat([self.sim.event_chains, row], ignore_index=True)
+                else:
+                    df_before = self.sim.population.props.copy()
+        
+        
         self.apply(self.target)
         self.post_apply_hook()
+                
+        if print_chains:
+            if self.target != self.sim.population:
+                row = self.sim.population.props.iloc[[self.target]]
+                row['person_ID'] = self.target
+                row['event'] = self
+                row['event_date'] = self.sim.date
+                row['when'] = 'After'
+                self.sim.event_chains = pd.concat([self.sim.event_chains, row], ignore_index=True)
```

Comment:
I believe that it's faster to not do these kinds of pandas operations very often, but instead to collect the data in python native structures (sets, dicts, lists, tuples) and then assemble them into a data-frame at the end.